sql/core/src/main/scala/org/apache/spark/sql/StandingQuery.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import org.apache.spark.annotation.DeveloperApi
+
+trait StandingQuery {
+
+  /**
+   * Stops the execution of the streaming query if it is running.  This method blocks until the
+   * thread performing execution has stopped.
+   */
+  def stop(): Unit
+
+  /** Clears the indicator that a batch has completed.  Used for testing. */
+  @DeveloperApi
+  def clearBatchMarker(): Unit
```

Comment:
For future note, we can extend it to 
`def awaitBatchAfter()(preBatchBody: => Unit)(postBatchBody: Batch => Unit)`

Then we can internally guarantee that the `preBatchBody` can be executed before a batch starts, and exactly one batch gets executed, and the `postBatchBody` is run. 
```
query.awaitBatchAfter {
    // Add some data
    stringToFile(new File(src, "1"), "drop1\nkeep2\nkeep3")
} { 
    // Test whether data has been received.
}
```